extension/src/ballistics/bullet/model.rs

```diff
@@ -0,0 +1,216 @@
+    pub air_friction: f64,
+    pub bullet_caliber: f64,
+    pub bullet_length: f64,
+    pub bullet_mass: f64,
+    pub ballistic_coefficients: Vec<f64>,
+    pub velocity_boundaries: Vec<f64>,
+    pub atmosphere_model: AtmosphereModel,
+    pub drag_function: DragFunction,
+    pub muzzle_velocities: Vec<f64>,
+    pub barrel_length: f64,
+    pub stability_factor: f64,
+    pub twist_direction: f64,
+    pub transonic_stability_coefficient: f64,
+    pub bullet_velocity_last_frame: Vector3,
+    pub origin: Vector3,
+    pub latitude: f64,
+    pub temperature: Temperature,
+    pub altitude: f64,
+    pub humidity: f64,
+    pub overcast: f64,
+    pub start_time: f64,
+    pub last_frame: f64,
+    pub rng: ChaCha8Rng,
+}
+
+impl Bullet {
+    pub fn simulate(
+        &mut self,
+        map: &Map,
+        bullet_velocity_current_frame: Vector3,
+        bullet_position: Vector3,
+        mut wind: Vector3,
+        height_atl: f64,
+        tick_time: f64,
+    ) -> Vector3 {
+        const EARTH_ANGULAR_SPEED: f64 = 0.000_072_92;
+        let mut tof = tick_time - self.start_time;
+        let delta_time = tick_time - self.last_frame;
+
+        let temperature =
+            Temperature::new_celsius(self.temperature.as_celsius() - 0.0065 * bullet_position.z());
+        let pressure = (1010.32 - 10.0 * self.overcast)
+            * (1.0
+                - (0.0065 * (self.altitude + bullet_position.z()))
+                    / 0.0065f64.mul_add(self.altitude, temperature.as_kelvin()))
+            .powf(5.255_754_495);
+
+        self.last_frame = tick_time;
+
+        if wind.magnitude() > 0.1 {
+            let mut wind_attenuation = 1.0;
+            let wind_source_terrain = bullet_position - wind.normalize() * 100.0;
+
+            let grid_x = (wind_source_terrain.x() / 50.0).floor() as i64;
+            let grid_y = (wind_source_terrain.y() / 50.0).floor() as i64;
+            let grid_cell = grid_x * map.grids() as i64 + grid_y;
+
+            if grid_cell >= 0 {
+                let grid_cell = grid_cell as usize;
+                if grid_cell < map.heights().len() && grid_cell < map.building_nums().len() {
+                    let grid_height = map.height(grid_cell) as f64;
+                    if grid_height > bullet_position.z() {
+                        let angle = ((grid_height - bullet_position.z()) / 100.0).atan();
+                        wind_attenuation *= angle.cos().abs().powi(2);
+                    }
+                }
+            }
+
+            if height_atl > 0.0 && height_atl < 20.0 {
+                let wind_source_obstacles = bullet_position - wind.normalize() * 25.0;
+                let roughness_length = calculate_roughness_length(
+                    map,
+                    wind_source_obstacles.x(),
+                    wind_source_obstacles.y(),
+                );
+                wind_attenuation *=
+                    0.0f64.max(height_atl / roughness_length.log(20.0 / roughness_length));
```

Comment:
[c++ src](https://github.com/acemod/ACE3/blob/master/extensions/advanced_ballistics/AdvancedBallistics.cpp#L533)
```suggestion
                wind_attenuation *= 0.0f64.max(
                    (height_atl / roughness_length).log10() / (20.0 / roughness_length).log10(),
                )
```
This is the one responsible for the drastic turns in trajectory

Reply:
![image](https://github.com/acemod/ACE3/assets/3946739/7f499e6f-86bd-4802-9054-aff984089057)
This was why I had made this change, Rust compilers tells me it is the same